### PR TITLE
Exposing default_mode of WAF Rule through cloudflare_waf_rules data source

### PR DESCRIPTION
**Pull request by JoseRoman**
#979

### CODE REVIEW

**Comment by jacobbednarz** [comment]
seems reasonable 👍 can you please add test coverage for this new attribute?

**Comment by JoseRoman** [comment]
@jacobbednarz Of course, I can add some tests for this.

- What is the process of running the acceptance tests locally? Do I have to create a test Cloudflare account to be able to run them? 
- This is a read-only field (similar to `allowed_modes`) that is not being filtered, do we still write tests for these?

**Comment by jacobbednarz** [comment]
> What is the process of running the acceptance tests locally? Do I have to create a test Cloudflare account to be able to run them?

There is a section in the README on [developing the provider](https://github.com/cloudflare/terraform-provider-cloudflare#developing-the-provider) which will step you through testing. Yes, you'll need a Cloudflare account as the tests create real resources.

> This is a read-only field (similar to allowed_modes) that is not being filtered, do we still write tests for these?

https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/data_source_waf_rules_test.go#L13 is an example of an existing `data_source_waf_rules` test which you could copy, tweak and confirm your new fields are usable (either through confirming them being written to the state file or in another resource).

File: cloudflare/data_source_waf_rules_test.go [hunk 1]
```diff
@@ -52,6 +52,27 @@ func TestAccCloudflareWAFRules_MatchDescription(t *testing.T) {
...
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareWAFRulesConfig(zoneID, map[string]string{}, rnd),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareWAFRulesDataSourceID(name),
+					resource.TestCheckResourceAttrSet(name, "rules.#"),
```
Comment (jacobbednarz) [inline cloudflare/data_source_waf_rules_test.go:69]: i'm afraid this assertion doesn't do assert anything useful for us here. this test is identical to some others and doesn't actually cover the functionality you've added. 

perhaps have a look at `testAccCheckCloudflareWAFRulesDataSourceID` and i suspect you'll need to programmatically loop through the resource rules and confirm the key is present in the state following this change.

**Review by jacobbednarz — CHANGES_REQUESTED** [review]
we need to adjust the test case for this because as it stands, it is a duplicate of an existing test with a different name and doesn't cover the added fields being stored in state.

**Comment by jacobbednarz** [comment]
closing in favour of #1079
